runtime/src/iree/hal/utils/libmpi_test.cc

```diff
@@ -88,4 +89,188 @@ TEST_F(LibmpiTest, MPIErrorToIREEStatus) {
               StatusIs(StatusCode::kInternal));
 }
 
-}  // namespace
+TEST_F(LibmpiTest, Bcast) {
+  const int N = 100;
+  char array[N];
+  int root = 1;
+
+  ASSERT_GT(world_size, 1)
+      << "This test requires mpirun with more than one node";
+
+  if (rank == root) memset(array, 'r', N);
+
+  IREE_EXPECT_OK(
+      MPI_RESULT_TO_STATUS(&symbols,
+                           MPI_Bcast(array, N, IREE_MPI_BYTE(&symbols), root,
+                                     IREE_MPI_COMM_WORLD(&symbols)),
+                           "MPI_Bcast"));
+
+  for (int i = 0; i < N; i++)
+    EXPECT_EQ(array[i], 'r') << "[" << rank << "] failed at index " << i;
+}
+
+TEST_F(LibmpiTest, Gather) {
+  const int N = 100;
+  char sendarray[N];
+  int root = 1;
+  char *recvbuf = NULL;
+
+  ASSERT_GT(world_size, 1)
+      << "This test requires mpirun with more than one node";
+
+  memset(sendarray, (char)(rank + 1), N);
+
+  if (rank == root) {
+    recvbuf = (char *)malloc(world_size * N * sizeof(char));
+    ASSERT_NE(recvbuf, nullptr);
+  }
+  IREE_EXPECT_OK(MPI_RESULT_TO_STATUS(
+      &symbols,
+      MPI_Gather(sendarray, N, IREE_MPI_BYTE(&symbols), recvbuf, N,
+                 IREE_MPI_BYTE(&symbols), root, IREE_MPI_COMM_WORLD(&symbols)),
+      "MPI_Gather"));
+
+  if (rank == root) {
+    for (int i = 0; i < N * world_size; i++) {
+      EXPECT_EQ(recvbuf[i], (char)((i / N) + 1))
+          << "[" << rank << "] failed at index " << i;
+    }
+    free(recvbuf);
+  }
+}
+
+TEST_F(LibmpiTest, Scatter) {
+  const int N = 100;
+  char recvbuf[N];
+  int root = 1;
+  char *sendbuf = NULL;
```

Comment:
style: asterisks on the left (here and elsewhere in the file)
```suggestion
  char* sendbuf = NULL;
```
(ensure your editor is configured to pick up the .clang-format file in the repo root or change your setting)

Reply:
I have been running clang-format manually from the repo root before committing the code and it keeps switching the pointers from left to right. I should think there is some mis-configuration in the rules for clang-format. 

The config file only specifies the base style, 'Google'.
My clang-format version is: `clang-format version google3-trunk (30af2fb33ed2f610abfa50e53df9712887b2bd25)`.
I don't have other ideas on where to look for what's mis-configured.

Reply:
Sometimes it infers based on the current file, so if you type them out on the right side it'll move all the others to the right side.

Reply:
Our config inherits the `DerivePointerAlignment` (https://clang.llvm.org/docs/ClangFormatStyleOptions.html#derivepointeralignment) setting, so clang-format will pick whatever style is most prevalent in the current file. Compiler code uses `char *someObject` (LLVM style) and runtime code uses `char* some_object` (Google C++ style).

I think we could make that setting explicit as part of https://github.com/openxla/iree/issues/12866

Reply:
+1 on making the left hand pointers explicit and remove the non-determinism. My personal style has always been the star on the right and it needs to be overridden :)